client/jetpack-cloud/sections/partner-portal/credit-card-fields/credit-card-element-field.tsx

```diff
@@ -0,0 +1,52 @@
+import { FormStatus, useFormStatus } from '@automattic/composite-checkout';
+import { CardElement } from '@stripe/react-stripe-js';
+import { StripeElementChangeEvent, StripeElementStyle } from '@stripe/stripe-js';
+import { useSelect } from '@wordpress/data';
+import { useI18n } from '@wordpress/react-i18n';
+import classnames from 'classnames';
+
+export default function CreditCardElementField( {
+	setIsStripeFullyLoaded,
+	handleStripeFieldChange,
+	stripeElementStyle,
+}: {
+	setIsStripeFullyLoaded: ( value: boolean ) => void;
+	handleStripeFieldChange: ( input: StripeElementChangeEvent ) => void;
+	stripeElementStyle: StripeElementStyle;
+} ) {
+	const { __ } = useI18n();
+	const { formStatus } = useFormStatus();
+	const isDisabled = formStatus !== FormStatus.READY;
+	const { card: cardError } = useSelect( ( select ) =>
+		select( 'credit-card' ).getCardDataErrors()
+	);
+
+	return (
+		<div className="credit-card-fields__input-field">
+			<label className="credit-card-fields__label">
+				<span className="credit-card-fields__label-text">{ __( 'Card details' ) }</span>
+				<span
+					className={ classnames( {
+						'credit-card-fields__stripe-element': true,
+						'credit-card-fields__stripe-element--has-error': cardError,
+						number: true,
+					} ) }
+				>
+					<CardElement
+						options={ {
+							disabled: isDisabled,
+							style: stripeElementStyle,
+						} }
+						onReady={ () => {
+							setIsStripeFullyLoaded( true );
+						} }
+						onChange={ ( input ) => {
+							handleStripeFieldChange( input );
+						} }
```

Comment:
Depending on how deep the tree of elements below this one is, one could suggest wrapping these functions with `useCallback` to avoid recreating them and the entire tree below them every time the component is rendered.

However, in this case, I don't think it is necessary since I don't think the tree is too deep to be worth it, but it is good to keep it in mind.